client/lib/mixins/url-search/component.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import { doSearch, getSearchOpen } from './index';
+
+export default function( Component ) {
+	const componentName = Component.displayName || Component.name || '';
+
+	return React.createClass( {
+		displayName: 'Searchable' + componentName,
+
+		getInitialState() {
+			return {
+				searchOpen: false
+			};
+		},
+
+		componentWillReceiveProps( nextProps ) {
+			if ( ! nextProps.search ) {
+				this.setState( {
+					searchOpen: false
+				} );
+			}
+		},
+
+		render() {
+			const props = assign( {}, this.props, {
+				doSearch: doSearch.bind( this ),
+				getSearchOpen: getSearchOpen.bind( this )
+			} );
+			return React.createElement( Component, props );
+		}
```

Comment:
Facebook doesn't appear to want to see `createElement` either if using JSX.

```js
render() {
	return (
		<Component { 
			...this.props, 
			doSearch: this.doSearch,
			getSearchOpen: this.getSearch,
		} />
	);
}
```

Note that I've changed the `bind()`ing here and will instead move it up into the class. The reason is that when we do this on every render, we actually create new bindings and totally new functions/closures on every render, which we don't want.